src/command/render/output.ts

```diff
@@ -129,7 +133,8 @@ export function outputRecipe(
     }
 
     // complete hook for keep-yaml
-    if (recipe.keepYaml) {
+    // workaround for https://github.com/quarto-dev/quarto-cli/issues/5079
+    if (recipe.keepYaml || recipe.format.render[kPreserveYaml]) {
```

Comment:
Regarding `preserve-yaml` addition, I believe we are round-tripping here. 

As discussed already in https://github.com/quarto-dev/quarto-cli/issues/4210, Tehre was a `keep-yaml` option, which was removed in 1.2 (detail here: https://github.com/quarto-dev/quarto-cli/issues/4210#issuecomment-1452185273)

The YAML preservation should be done by using the Pandoc's extension  `+yaml_metadata_block`
Adding this variant to the format extension should be enough. Did you try that already ?

I  am trying to keep things clear here and helping identify the correct fix.

Reply:
You're not correct, but it's easier to explain on Slack. Let me know there if you want the full story.